### ISSUE TITLE
azurerm_container_app_job allow replica_retry_limit to be equal to 0

### ISSUE THREAD

**Issue by nathan-mittelette** [hashicorp/terraform-provider-azurerm#25983]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.8.3

### AzureRM Provider Version

3.103.1

### Affected Resource(s)/Data Source(s)

azurerm_container_app_job

### Terraform Configuration Files

```hcl
resource "azurerm_container_app_job" "acaj" {
  name                         = "${local.application_name_full}-acaj"
  container_app_environment_id = var.container_app_env_id
  location                     = var.location
  resource_group_name          = var.rg_name
  tags                         = var.tags

  replica_timeout_in_seconds = var.replica_timeout_in_seconds
  replica_retry_limit        = 0 // HERE
  schedule_trigger_config {
      cron_expression          = schedule_trigger_config.value.cron_expression
      parallelism              = schedule_trigger_config.value.parallelism
      replica_completion_count = schedule_trigger_config.value.replica_completion_count
  }

  // Configure the container
  template {
    container {
      // Setting the container image and resources
      name    = var.container_name
      image   = "${var.container_image}:${var.container_tag}"
      args    = var.container_args
      command = var.container_command
      cpu     = var.container_cpu
      memory  = var.container_memory
    }
  }
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: expected replica_retry_limit to be at least (1), got 0
│
```

### Expected Behaviour

Actually, it's not possible to provide the value `0` to the attribute `replica_retry_limit` in the resource `azurerm_container_app_job`.

In the Azure Documentation, it's write that is possible to provide the value 0 for the configuration. We can also provide the value directly from the website.

Link from the documentation : [here](https://learn.microsoft.com/en-us/azure/container-apps/jobs?tabs=azure-cli#job-settings)

<img width="849" alt="Capture d’écran 2024-05-15 à 19 01 25" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/52243730/d50fe167-21c1-4cd6-9539-6fcbb841c8bf">

### Actual Behaviour

When we provide the value `0` to the attribute `replica_retry_limit` we've got an validating error : `Error: expected replica_retry_limit to be at least (1), got 0`

### Steps to Reproduce

1. terraform apply

**Comment by Chambras** [comment]
@nathan-mittelette nice finding! I think I know where the issue is. I will try to submit a PR soon.

**Comment by nathan-mittelette** [comment]
@Chambras, I've already started a pull request with this small update. I hope this saves you some time. Please take a look when you have a moment. #25984

**Comment by Chambras** [comment]
yeah, that looks good. I think the reason it was not caught it was because there isn't a test case for that scenario...
